standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/ReplChangeManager.java

```diff
@@ -156,13 +157,21 @@ private ReplChangeManager(Configuration conf) throws MetaException {
           //Create default cm root
           Path cmroot = new Path(cmRootDir);
           createCmRoot(cmroot);
+          FileSystem cmRootFs = cmroot.getFileSystem(conf);
           HdfsEncryptionShim pathEncryptionShim = hadoopShims
-                  .createHdfsEncryptionShim(cmroot.getFileSystem(conf), conf);
+                  .createHdfsEncryptionShim(cmRootFs, conf);
           if (pathEncryptionShim.isPathEncrypted(cmroot)) {
```

Comment:
if cmroot is encrypted ..it should throw exception ..

Reply:
At recycle we will throw an exception if ever a non encrypted path is passed.